core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableHashJoin.java

```diff
@@ -56,8 +55,7 @@ protected EnumerableHashJoin(
       RelNode right,
       RexNode condition,
       Set<CorrelationId> variablesSet,
-      JoinRelType joinType)
-      throws InvalidRelException {
+      JoinRelType joinType) {
```

Comment:
Would it make sense to add in this constructor a check on the condition (to verify that it is actually an equi-join, which is required by the underlying implementation algorithm)?
```
if (!JoinInfo.of(left, right, condition).isEqui())
  throw new IllegalArgumentException("...");
```
or maybe at least an assert
`assert JoinInfo.of(left, right, condition).isEqui();`

Reply:
Yes, make sense. Will do